floodviz/hydrograph_utils.py

```diff
@@ -37,8 +41,27 @@ def parse_hydrodata(jdata):
                 dt = datetime.strptime(date + ' ' + t, '%Y-%m-%d %H:%M:%S')
                 # Convert to milliseconds for use with d3 x axis format
                 dt_ms = time.mktime(dt.timetuple()) * 1000
+                # handle missing data
+                if prev_date_ms:
+                    # Size of potential gap in data
+                    gap = dt_ms - prev_date_ms
+                    """
+                      We only consider data to be missing if the gap between adjacent data points 
+                      is greater than 30 minutes. All dummy data points will be added in 15 minute increments.
+                    """
+                    if gap > gap_threshold:
+                        num_dummy_points = gap/increment_ms - 1
+                        added = 0
+                        while added < num_dummy_points:
+                            # correct time for this dummy point
+                            new_dt_ms = prev_date_ms + (increment_ms * (added + 1))
+                            all_series_data.append({'key': site_id, 'name': 'no_data', 'date': date, "time": t,
```

Comment:
You need to find another way to do the test then. The siteName should be the siteName whether there is data or not.